devito/ir/equations/equation.py

```diff
@@ -158,13 +168,15 @@ def __new__(cls, *args, **kwargs):
                  for k, v in mapper.items() if k}
         dspace = DataSpace(dintervals, parts)
 
-        # Construct the conditionals
+        # Construct the conditionals and replace the ConditionalDimensions in `expr`
         conditionals = {}
         for d in conditional_dimensions:
             if d.condition is None:
                 conditionals[d] = CondEq(d.parent % d.factor, 0)
             else:
                 conditionals[d] = diff2sympy(lower_exprs(d.condition))
+            if d.factor is not None:
+                expr = uxreplace(expr, {d: IntDiv(d.index, d.factor)})
```

Comment:
what cases is this needed for?

Reply:
it's the result of lifting [this routine](https://github.com/devitocodes/devito/blob/master/devito/ir/iet/scheduler.py#L95) from the guts of the compiler up here. So it's an improvement (logically, this is where this sort of lowering should be performed)